Wikipedia/Code/TalkPageViewController.swift

```diff
@@ -355,9 +355,19 @@ extension TalkPageViewController: TalkPageCellDelegate {
         }
         
         let configuredCellViewModel = viewModel.topics[indexOfConfiguredCell]
-        configuredCellViewModel.isSubscribed.toggle()
-        cell.configure(viewModel: configuredCellViewModel, linkDelegate: self)
-        self.handleSubscriptionAlert(isSubscribedToTopic: configuredCellViewModel.isSubscribed)
+        
+        let shouldSubscribe = !configuredCellViewModel.isSubscribed
+        viewModel.subscribe(to: configuredCellViewModel.topicName, shouldSubscribe: shouldSubscribe) { result in
+            switch result {
+            case let .success(didSubscribe):
+                cell.configure(viewModel: configuredCellViewModel, linkDelegate: self)
+                self.handleSubscriptionAlert(isSubscribedToTopic: didSubscribe)
+            case let .failure(error):
+                DDLogError("Error subscribing to topic: \(error)")
+                //TODO: Error handling
+            }
+        }
+ 
     }
```

Comment:
This is my thinking for this method. Without this approach, the button won't change states immediately with a slow Internet connection.

```
        guard let cellViewModel = cellViewModel else {
            return
        }
        
        let shouldSubscribe = cellViewModel.isSubscribed
        
        // Optimistically switching cell state. It will revert later if the call fails.
        cellViewModel.isSubscribed.toggle()
        cell.configure(viewModel: cellViewModel, linkDelegate: self)
        
        viewModel.subscribe(to: cellViewModel.topicName, shouldSubscribe: shouldSubscribe) { result in
            switch result {
            case let .success(didSubscribe):
                self.handleSubscriptionAlert(isSubscribedToTopic: didSubscribe)
            case let .failure(error):
                DDLogError("Error subscribing to topic: \(error)")
                
                // Revert isSubscribed state back
                cellViewModel.isSubscribed.toggle()
                
                // Reconfigure cell, if it already hasn't been reconfigured with another cell view model
                if cell.viewModel?.topicName == cellViewModel.topicName {
                    cell.configure(viewModel: cellViewModel, linkDelegate: self)
                }
                
                // TODO: Error banner
            }
        }
```

Reply:
Good point; it does make it smoother! I'm just curious about this check `if cell.viewModel?.topicName == cellViewModel.topicName {..}`, do you think there's a risk to configuring it with the incorrect view model?

Reply:
Yep - I'm worried here that the cell could already be reused elsewhere in the collection view once the POST call completes, so I added that check. Although that thinking could be outdated with collection view / table view changes in recent years - I did some testing this morning on a similar concern elsewhere and never ran into it.

A user could subscribe, the button state immediately changes to subscribed, then they scroll down for a while as the API call goes through on a slow connection. If they have scrolled down far enough for this cell to get reused and receive a different topic view model in this interim period, we want to NOT reconfigure it with this old failed subscription view model. Without this check, they would suddenly see that same failed subscription topic from above appear at the bottom of the collection view. Hope that makes sense!

Reply:
It makes sense, thanks for clarifying that! I added this verification, I think it's ready for another look